spec/webauthn/public_key_credential_spec.rb

```diff
@@ -92,5 +92,68 @@
         }.to raise_error(WebAuthn::ChallengeVerificationError)
       end
     end
+
+    context "when clientExtensionResults is received" do
+      let(:public_key_credential) do
+        WebAuthn::PublicKeyCredentialWithAttestation.new(
+          type: type,
+          id: id,
+          raw_id: raw_id,
+          client_extension_outputs: { "appid" => "true" },
+          response: attestation_response
+        )
+      end
+
+      it "works" do
+        expect(public_key_credential.verify(challenge)).to be_truthy
+
+        expect(public_key_credential.client_extension_outputs).to eq({ "appid" => "true" })
+      end
+    end
+
+    context "when authentication extension input" do
+      context "is not received" do
+        let(:attestation_response) do
+          allow_any_instance_of(WebAuthn::FakeAuthenticator::AuthenticatorData)
+            .to receive(:extensions)
+            .and_return(nil)
```

Comment:
The following discussion was accidentally deleted by me in an attempt to separate different discussions into different threads – originally it was placed after [this comment](https://github.com/cedarcode/webauthn-ruby/pull/317/files#r434550481), in the same thread.
Turns out that if you delete the first message of a thread, you'll actually delete the whole tread 🙃  
I was able to re-create it thanks to @marceloeloelo 🥳

Reply:
_The initial comment was:_

Is not that clear to me if in this case extensions should return `nil` or `{}` 😕

I don't think is possible that one can initialize an `WebAuthn::FakeAuthenticator::AuthenticatorData` with `extensions` being `nil`, given that it has [`{ "fakeExtension" => "fakeExtensionValue" }` as the default value](https://github.com/cedarcode/webauthn-ruby/blob/a4bca13c8f6cf9d24ea10b3b6ca9bae8a9a53a63/lib/webauthn/fake_authenticator/authenticator_data.rb#L24).

But, on the other hand, the test fails if we use {} as blank authentication extension input, given that the fake authenticator will consider that extension data is received – [`extension_data`](https://github.com/cedarcode/webauthn-ruby/blob/a4bca13c8f6cf9d24ea10b3b6ca9bae8a9a53a63/lib/webauthn/fake_authenticator/authenticator_data.rb#L74:L80) will encode in CBOR the object `{}` and later [`extension_data_included_bit`](https://github.com/cedarcode/webauthn-ruby/blob/a4bca13c8f6cf9d24ea10b3b6ca9bae8a9a53a63/lib/webauthn/fake_authenticator/authenticator_data.rb#L98:L104) will return `1`.
So, a little tweak has to be made to this model if we agree that we should consider `{}` as blank exceptions.

To avoid doing changes to the fake authenticator data I just went with the first one but I would say that we should consider going for the second one and do that little refactor.

Reply:
_To which @bdewater replied:_

You can initialize an `WebAuthn::FakeAuthenticator::AuthenticatorData` with `extensions: nil`. It's only when the extensions key is not specified by the caller the default value is used. You can think of of Ruby keyword arguments as a hash with default values being merged with caller values :)

Based on my reading of CTAP 2.0 the current behaviour seems correct, extensions are optional and it's presence indicates means they should be processed. It seems it is unlikely the result will be the equivalent of an empty Ruby hash in real life 🤔

Reply:
_Finally, my response was:_

>You can initialize an WebAuthn::FakeAuthenticator::AuthenticatorData with extensions: nil. It's only when the extensions key is not specified by the caller the default value is used. You can think of of Ruby keyword arguments as a hash with default values being merged with caller values :)

Oh I see, that makes sense! Thanks for the clarification! 🙂 

>Based on my reading of CTAP 2.0 the current behaviour seems correct, extensions are optional and it's presence indicates means they should be processed. It seems it is unlikely the result will be the equivalent of an empty Ruby hash in real life 🤔

Good to know! Thanks for clarifying that! I supposed that the current behavior was correct that's why I was hesitant to make that refactor without consulting to you guys first 😄